### ISSUE TITLE
Allow users to use cloudfront for AWS S3 bucket

### ISSUE THREAD

**Issue by mazharoddin** [owen2345/camaleon-cms#291]
Hi Owen,
Can you provide option to use cloudfront for AWS S3 bucket assets(images uploaded on posts), Images are taking long time to load from AWS S3 bucket. With carrierwave gem I used to create a cloudfront distribution for AWS S3 bucket( separate distribution for images/videos etc, its origin will be xxx.s3.amazonaws.com as opposed to site wide distribution for static assets) and used to configure carrierwave initializer as below. After this all images which are uploaded to S3 will get cached on cloudfront. This improves overall performance and reduces image download time considerably.
````  
config/initializers/carrierwave.rb
  config.asset_host = "http://xxxx.cloudfront.net"
````
Request you to provide similar option in Camaleon CMS. This settings can go along with S3 settings in filesystem settings menu option.

**Comment by mazharoddin** [comment]
Hi @owen2345, please look into this issue if you got some time, images from S3 bucket taking lot of time to load without cloudfront CDN. 
Please also refer to #280 issue for cloudfront related discussion.

**Comment by owen2345** [comment]
@mazharoddin 
Could you please give a sample to integrate cloudfront?
actually I removed the fog dependency and I added custom uploaders for some for some problems with Fog_local (Same functionality to Fog).
Tips:
- When I upload a file to AWS, it returns me and AWS-Object, that includes public_url
- Do I need to replace the url?
Sample aws url: 
```
https://s3-us-west-2.amazonaws.com/owen/my_image.png
```
into 
```
http://xxxx.cloudfront.net/owen/my_image.png
```

Regards,

**Comment by owen2345** [comment]
Ignore previous comment, it was solved in #280
